src/lavinmq/client/channel/consumer.cr

```diff
@@ -241,6 +241,39 @@ module LavinMQ
 
         class ClosedError < Error; end
       end
+
+      class StreamConsumer < LavinMQ::Client::Channel::Consumer
+        @segment = 1_u32
+        @offset = nil
+        @pos = 4_u32
+        @requeued = Deque(SegmentPosition).new
+        getter offset, segment, pos, requeued
+
+        def initialize(@channel : Client::Channel, @tag : String,
+                       @queue : Queue, @no_ack : Bool, @exclusive : Bool, @priority : Int32, @offset : Int64?)
+          super(@channel, @tag, @queue, @no_ack, @exclusive, @priority)
+        end
+
+        def update_segment(segment, pos)
+          @segment = segment
+          @pos = pos
+        end
+
+        def update_offset(offset)
+          @offset = offset
+        end
+
+        def reject(unack, requeue)
+          was_full = @unacked == @prefetch_count
+          @unacked -= 1
+          notify_hash_capacity(true) if was_full
```

Comment:
no need to notify the queue, because the size of the queue hasn't changed, other stream consumers aren't intreseted in this